### ISSUE TITLE
mastering-display

### ISSUE THREAD

**Issue by jriker1** [BlueSwordM/aom-av1-psy#6]
Know I guess you moved on from this to rav1e which is a shame, looks fairly extensive what was implemented.  That said am I missing the option to apply mastering-display and content-light info?

**Comment by BlueSwordM** [comment]
Yes. mainline aomenc doesn't have access to this, and so this doesn't.

**Comment by jriker1** [comment]
Thanks, this is a bummer.  Think I'm giving up on AV1 for now.  Just not ready for my needs yet.  That said, any idea why 232 frames into encoding a 4k HDR10 Blu-ray video using the below commands, it's already at 3.5gb?  Think it will be 2.5TB in size if I encoded the whole video:

`aom-av1-psy-Full_Build-Generic-LTO-2022-09-06.exe --passes=2 --ivf --i420 --threads=30 --width=3840 --height=2160 --fps=24/1 --bit-depth=10 --input-bit-depth=10 --end-usage=q --cpu-used=0 --cq-level=16 --kf-min-dist=12 --kf-max-dist=240 --enable-chroma-deltaq=1 --quant-b-adapt=1 --enable-qm=1 --min-q=1 --enable-keyframe-filtering=0 --arnr-strength=1 --sharpness=3 --enable-dnl-denoising=0 --denoise-noise-level=5 --disable-trellis-quant=0 --tune-content=psy --tune=psnr --deltaq-mode=5 --color-primaries=bt2020 --transfer-characteristics=smpte2084 --matrix-coefficients=bt2020ncl --chroma-sample-position=colocated --set-tier-mask=0 --vmaf-model-path=vmaf_v0.6.1.json --limit=360 -o <output>-AOM.ivf <input>.mkv`

**Comment by BlueSwordM** [comment]
Well, aomenc doesn't support matroska input, so you are likely writing to y4m :)

**Comment by jriker1** [comment]
Wait, if it doesn't accept Matroska input, then wouldn't it fail to start encoding?  I could see that being the answer if I was doing that kind of output vs input.

Note the final output was garbage
![image](https://user-images.githubusercontent.com/1495536/227585040-ab6a0bd8-5b03-4967-9e1f-1ad2f8349ea7.png)

but as mentioned odd it would output to y4m if the input was the wrong format.  Going to have to try again since looks like it doesn't error out with bad input files.  Better to pipe thru ffmpeg to convert to y4m first?  Or is there another format it accepts?  I just stuck it in that container doesn't have to be.


By the way, how do you do two pass with piping the source if that's the issue?  I tried the below which is my current final command so open to input and did one pass and then the second pass completed instantly with frame 0/0:

`ffmpeg.exe -hide_banner -loglevel quiet -t 30 -i <input>.mkv -an -pix_fmt yuv420p10le -strict -1 -f yuv4mpegpipe - | aom-av1-psy-Full_Build-Generic-LTO-2022-09-06.exe --passes=2 --lag-in-frames=48 --ivf --i420 --threads=30 --tile-columns=2 --tile-rows=1 --width=3840 --height=2160 --fps=24/1 --bit-depth=10 --input-bit-depth=10 --end-usage=q --cpu-used=2 --cq-level=16 --kf-min-dist=12 --kf-max-dist=240 --enable-chroma-deltaq=0 --quant-b-adapt=1 --enable-qm=1 --min-q=1 --enable-keyframe-filtering=0 --arnr-strength=1 --arnr-maxframes=3 --sharpness=3 --enable-dnl-denoising=0 --denoise-noise-level=5 --disable-trellis-quant=0 --tune-content=psy --tune=psnr --enable-fwd-kf=1 --deltaq-mode=5 --color-primaries=bt2020 --transfer-characteristics=smpte2084 --matrix-coefficients=bt2020ncl --enable-restoration=0 --chroma-sample-position=colocated --set-tier-mask=0 --vmaf-model-path=vmaf_v0.6.1.json -o  <output>-AOM.ivf -`

**Comment by BlueSwordM** [comment]
I use av1an which automates 2-pass stuff.
